cmd/frontend/graphqlbackend/access_requests.go

```diff
@@ -0,0 +1,148 @@
+		db:   r.db,
+		args: &args.AccessRequestsFilterArgs,
+	}
+
+	reverse := false
+	connectionOptions := graphqlutil.ConnectionResolverOptions{
+		Reverse:   &reverse,
+		OrderBy:   database.OrderBy{{Field: string(database.AccessRequestListID)}},
+		Ascending: false,
+	}
+	return graphqlutil.NewConnectionResolver[*accessRequestResolver](connectionStore, &args.ConnectionResolverArgs, &connectionOptions)
+}
+
+type accessRequestConnectionStore struct {
+	db   database.DB
+	args *database.AccessRequestsFilterArgs
+}
+
+func (s *accessRequestConnectionStore) ComputeTotal(ctx context.Context) (*int32, error) {
+	count, err := s.db.AccessRequests().Count(ctx, s.args)
+	if err != nil {
+		return nil, err
+	}
+
+	totalCount := int32(count)
+
+	return &totalCount, nil
+}
+
+func (s *accessRequestConnectionStore) ComputeNodes(ctx context.Context, args *database.PaginationArgs) ([]*accessRequestResolver, error) {
+	accessRequests, err := s.db.AccessRequests().List(ctx, s.args, args)
+	if err != nil {
+		return nil, err
+	}
+
+	resolvers := make([]*accessRequestResolver, len(accessRequests))
+	for i, accessRequest := range accessRequests {
+		resolvers[i] = &accessRequestResolver{accessRequest: accessRequest}
+	}
+
+	return resolvers, nil
+}
+
+func (s *accessRequestConnectionStore) MarshalCursor(node *accessRequestResolver, _ database.OrderBy) (*string, error) {
+	if node == nil {
+		return nil, errors.New(`node is nil`)
+	}
+
+	cursor := string(node.ID())
+
+	return &cursor, nil
+}
+
+func (s *accessRequestConnectionStore) UnmarshalCursor(cursor string, _ database.OrderBy) (*string, error) {
+	nodeID, err := unmarshalAccessRequestID(graphql.ID(cursor))
+	if err != nil {
+		return nil, err
+	}
+
+	id := strconv.Itoa(int(nodeID))
+
+	return &id, nil
+}
+
+// accessRequestResolver resolves an access request.
+type accessRequestResolver struct {
+	accessRequest *types.AccessRequest
+}
+
+func (s *accessRequestResolver) ID() graphql.ID { return marshalAccessRequestID(s.accessRequest.ID) }
+
+func (s *accessRequestResolver) Name() string { return s.accessRequest.Name }
+
+func (s *accessRequestResolver) Email() string { return s.accessRequest.Email }
+
+func (s *accessRequestResolver) CreatedAt() gqlutil.DateTime {
+	return gqlutil.DateTime{Time: s.accessRequest.CreatedAt}
+}
+
+func (s *accessRequestResolver) AdditionalInfo() *string { return &s.accessRequest.AdditionalInfo }
+
+func (s *accessRequestResolver) Status() string { return string(s.accessRequest.Status) }
+
+func (r *schemaResolver) SetAccessRequestStatus(ctx context.Context, args *struct {
+	ID     graphql.ID
+	Status types.AccessRequestStatus
+}) (*EmptyResponse, error) {
+	// 🚨 SECURITY: Only site admins can update access requests.
+	if err := auth.CheckCurrentUserIsSiteAdmin(ctx, r.db); err != nil {
+		return nil, err
+	}
+
+	id, err := unmarshalAccessRequestID(args.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	store, err := r.db.AccessRequests().Transact(ctx)
```

Comment:
Nitpick, but this is usually what we do and it helps spot transactions on first read :)

```suggestion
	tx, err := r.db.AccessRequests().Transact(ctx)
```